--- a/lib/repoman/__init__.py	2021-03-28 10:23:28.088698000 +0800
+++ b/lib/repoman/__init__.py	2025-03-09 13:21:09.517414571 +0800
@@ -1,13 +1,14 @@
 
 import os.path
 import subprocess
+import shlex
 import sys
 import time
 
 try:
 	import portage.const
 	import portage.proxy as proxy
-	from portage import _encodings, _shell_quote, _unicode_encode, _unicode_decode
+	from portage import _encodings, _unicode_encode, _unicode_decode
 	from portage.const import PORTAGE_BASE_PATH, BASH_BINARY
 except ImportError as e:
 	sys.stderr.write("\n\n")
@@ -38,7 +39,7 @@
 				cmd = [BASH_BINARY, "-c", ("cd %s ; git describe  --match 'repoman-*' || exit $? ; " + \
 					"if [ -n \"`git diff-index --name-only --diff-filter=M HEAD`\" ] ; " + \
 					"then echo modified ; git rev-list --format=%%ct -n 1 HEAD ; fi ; " + \
-					"exit 0") % _shell_quote(PORTAGE_BASE_PATH)]
+					"exit 0") % shlex.quote(PORTAGE_BASE_PATH)]
 				cmd = [_unicode_encode(x, encoding=encoding, errors='strict')
 					for x in cmd]
 				proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
--- a/lib/repoman/modules/vcs/bzr/status.py	2021-03-28 10:23:28.092697900 +0800
+++ b/lib/repoman/modules/vcs/bzr/status.py	2025-03-09 13:26:12.065463136 +0800
@@ -2,6 +2,8 @@
 Bazaar module Status class submodule
 '''
 
+import shlex
+
 from repoman._portage import portage
 from portage import os
 
@@ -32,7 +34,7 @@
 		try:
 			myf = repoman_popen(
 				"bzr ls -v --kind=file " +
-				portage._shell_quote(checkdir))
+				shlex.quote(checkdir))
 			myl = myf.readlines()
 			myf.close()
 		except IOError:
--- a/lib/repoman/modules/vcs/cvs/changes.py	2021-03-28 10:23:28.092697900 +0800
+++ b/lib/repoman/modules/vcs/cvs/changes.py	2025-03-09 13:26:55.073764176 +0800
@@ -4,6 +4,7 @@
 
 import re
 from itertools import chain
+import shlex
 
 from repoman._portage import portage
 from repoman.modules.vcs.changes import ChangesBase
@@ -91,7 +92,7 @@
 				continue
 
 			_out = repoman_getstatusoutput(
-				"egrep -q %s %s" % (headerstring, portage._shell_quote(_file)))
+				"egrep -q %s %s" % (headerstring, shlex.quote(_file)))
 			if _out[0] == 0:
 				headers.append(_file)
 
--- a/lib/repoman/modules/vcs/git/status.py	2021-03-28 10:23:28.092697900 +0800
+++ b/lib/repoman/modules/vcs/git/status.py	2025-03-09 13:24:20.169883355 +0800
@@ -3,6 +3,7 @@
 '''
 
 import re
+import shlex
 
 from repoman._portage import portage
 from portage import os
@@ -33,7 +34,7 @@
 		'''
 		with repoman_popen(
 			"git ls-files --others %s" %
-			(portage._shell_quote(checkdir_relative),)) as myf:
+			(shlex.quote(checkdir_relative),)) as myf:
 			for l in myf:
 				if l[:-1][-7:] == ".ebuild":
 					self.qatracker.add_error(
--- a/lib/repoman/modules/vcs/hg/status.py	2021-03-28 10:23:28.092697900 +0800
+++ b/lib/repoman/modules/vcs/hg/status.py	2025-03-09 13:25:47.476434477 +0800
@@ -2,6 +2,8 @@
 Mercurial module Status class submodule
 '''
 
+import shlex
+
 from repoman._portage import portage
 from portage import os
 
@@ -31,7 +33,7 @@
 		'''
 		myf = repoman_popen(
 			"hg status --no-status --unknown %s" %
-			(portage._shell_quote(checkdir_relative),))
+			(shlex.quote(checkdir_relative),))
 		for l in myf:
 			if l[:-1][-7:] == ".ebuild":
 				self.qatracker.add_error(
--- a/lib/repoman/modules/vcs/svn/changes.py	2021-03-28 10:23:28.092697900 +0800
+++ b/lib/repoman/modules/vcs/svn/changes.py	2025-03-09 13:23:19.772269218 +0800
@@ -3,6 +3,7 @@
 '''
 
 from itertools import chain
+import shlex
 
 from repoman.modules.vcs.changes import ChangesBase
 from repoman._subprocess import repoman_popen
@@ -113,7 +114,7 @@
 			headerstring = r"'\$(%s).*\$'" % "|".join(enabled_keywords)
 
 			_out = repoman_getstatusoutput(
-				"egrep -q %s %s" % (headerstring, portage._shell_quote(_file)))
+				"egrep -q %s %s" % (headerstring, shlex.quote(_file)))
 			if _out[0] == 0:
 				headers.append(_file)
 
--- a/lib/repoman/modules/vcs/svn/status.py	2021-03-28 10:23:28.092697900 +0800
+++ b/lib/repoman/modules/vcs/svn/status.py	2025-03-09 13:22:02.945304053 +0800
@@ -3,6 +3,7 @@
 '''
 
 import logging
+import shlex
 import subprocess
 import sys
 
@@ -39,7 +40,7 @@
 		try:
 			myf = repoman_popen(
 				"svn status --depth=files --verbose " +
-				portage._shell_quote(checkdir))
+				shlex.quote(checkdir))
 			myl = myf.readlines()
 			myf.close()
 		except IOError:
@@ -56,7 +57,7 @@
 		try:
 			myf = repoman_popen(
 				"svn status " +
-				portage._shell_quote(checkdir))
+				shlex.quote(checkdir))
 			myl = myf.readlines()
 			myf.close()
 		except IOError:
--- a/lib/repoman/repos.py	2021-03-29 02:43:50.725197000 +0800
+++ b/lib/repoman/repos.py	2025-03-09 13:20:29.099011188 +0800
@@ -4,6 +4,7 @@
 import io
 import logging
 import re
+import shlex
 import sys
 import textwrap
 
@@ -165,7 +166,7 @@
 		# or PORTDIR_OVERLAY settings.
 		self.repoman_settings['PORTDIR_OVERLAY'] = "%s %s" % (
 			self.repoman_settings.get('PORTDIR_OVERLAY', ''),
-			portage._shell_quote(portdir_overlay))
+			shlex.quote(portdir_overlay))
 		self.repositories = self.repo_conf.load_repository_config(
 			self.repoman_settings, extra_files=[tmp_conf_file])
 		# We have to call the config constructor again so that attributes
